src/Core/Services/PasswordGenerationService.cs

```diff
@@ -255,7 +259,129 @@ public async Task<PasswordGenerationOptions> GetOptionsAsync()
+
+            var enforcedPolicyOptions = await GetPasswordGeneratorPolicyOptions();
+            if(enforcedPolicyOptions != null)
+            {
+                if(_optionsCache.Length < enforcedPolicyOptions.MinLength)
+                {
+                    _optionsCache.Length = enforcedPolicyOptions.MinLength;
+                }
+
+                if(enforcedPolicyOptions.UseUppercase)
+                {
+                    _optionsCache.Uppercase = true;
+                }
+
+                if(enforcedPolicyOptions.UseLowercase)
+                {
+                    _optionsCache.Lowercase = true;
+                }
+
+                if(enforcedPolicyOptions.UseNumbers)
+                {
+                    _optionsCache.Number = true;
+                }
+
+                if(_optionsCache.MinNumber < enforcedPolicyOptions.NumberCount)
+                {
+                    _optionsCache.MinNumber = enforcedPolicyOptions.NumberCount;
+                }
+
+                if(enforcedPolicyOptions.UseSpecial)
+                {
+                    _optionsCache.Special = true;
+                }
+
+                if(_optionsCache.MinSpecial < enforcedPolicyOptions.SpecialCount)
+                {
+                    _optionsCache.MinSpecial = enforcedPolicyOptions.SpecialCount;
+                }
+
+                // Must normalize these fields because the receiving call expects all options to pass the current rules
+                if(_optionsCache.MinSpecial + _optionsCache.MinNumber > _optionsCache.Length)
+                {
+                    _optionsCache.MinSpecial = _optionsCache.Length - _optionsCache.MinNumber;
+                }
+            }
+            else
+            {
+                // UI layer expects an instantiated object to prevent more explicit null checks
+                enforcedPolicyOptions = new PasswordGeneratorPolicyOptions();
+            }
+
+            return (_optionsCache, enforcedPolicyOptions);
+        }
+
+        public async Task<PasswordGeneratorPolicyOptions> GetPasswordGeneratorPolicyOptions()
+        {
+            var policies = await _policyService.GetAll(PolicyType.PasswordGenerator);
+            PasswordGeneratorPolicyOptions enforcedOptions = null;
+
+            if(policies == null || !policies.Any())
+            {
+                return enforcedOptions;
+            }
+
+            foreach(var currentPolicy in policies)
+            {
+                if(!currentPolicy.Enabled || currentPolicy.Data == null)
+                {
+                    continue;
+                }
+
+                if(enforcedOptions == null)
+                {
+                    enforcedOptions = new PasswordGeneratorPolicyOptions();
+                }
+
+                var currentPolicyMinLength = currentPolicy.Data["minLength"];
+                if(currentPolicyMinLength != null &&
+                   Convert.ToInt32(currentPolicyMinLength) > enforcedOptions.MinLength)
```

Comment:
I think just casting to int would be enough here, and a little cleaner. `(int)currentPolicyMinLength`.

`Convert` is more for when _changing_ from one type to another, like converting a string `"1"` to an int `1`.

Reply:
The object was coming back as an Int64 and throwing InvalidCastException.  Additional research shows double-unboxing will also work so I'll use that approach.